### PR TITLE
fix: useCustomResource hook infinite call

### PR DESCRIPTION
**Pull request by mclarke47**
Signed-off-by: Matthew Clarke <mclarke@spotify.com>

Fixes https://github.com/backstage/backstage/issues/13832

- [ ] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [ ] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: plugins/kubernetes/src/hooks/useCustomResources.ts [hunk 1]
```diff
@@ -40,7 +40,7 @@ export const useCustomResources = (
 ): KubernetesObjects => {
   const kubernetesApi = useApi(kubernetesApiRef);
   const kubernetesAuthProvidersApi = useApi(kubernetesAuthProvidersApiRef);
-
+  const matchersString = JSON.stringify(customResourceMatchers)
```
Comment (mclarke47) [inline plugins/kubernetes/src/hooks/useCustomResources.ts:43]: Tell me there is a better way....

**Comment by mclarke47** [comment]
I coudn't replicate this in a test which was really annoying

**Comment by mclarke47** [comment]
@Rugvip this is ready for review

**Review by Rugvip — APPROVED** [review]
Alright! 👍

**Pull request merged**